Stefan.nikolic tests pr

stefannikolic opened the pull request:
Stefan Nikolic first PR

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,35 @@
+describe('Login Test', () => {
+  let user: any;
```
We should add TypeScript interface (data model) for **user** type.

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,35 @@
+      user = data;
+    });
+  });
+  beforeEach(() => {
```
For better readability, we should separate each block with a new line.

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,35 @@
+    });
+  });
+  beforeEach(() => {
+    cy.visit('https://release-13-3-0--dib-travel.netlify.app/');
```
We should specify the URL in the config and communicate with the whole team similar change in the future.

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,35 @@
+describe('Login Test', () => {
+  let user: any;
+  before(() => {
+    cy.fixture('stefan').then((data) => {
```
You should use a descriptive name for a fixture - the file name should explain the content of the file.

sjova reviewed (commented):
Try to finalize registration tests before tomorrow's call.

And you didn't remove `.gitkeep` :)

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,43 @@
+import { StefanLog } from '../../../models';
+
+describe('Login Test', () => {
+  let user: StefanLog;
```
It's nice to add more context in a variable name. Also, use more descriptive names for the data model.

Ex. `let loginDetails: Login;` or `let loginTestCases: LoginTestCases;` or something similar.

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,43 @@
+  let user: StefanLog;
+
+  before(() => {
+    cy.fixture('stefan_nikolic_data').then((data) => {
```
Please use a descriptive name for fixture name.

dusan011 replied:
It would be good to name model in a better way(eg. loginDetails or something like that, not StefanLog)

sjova replied:
Just a minder that file should be dash-case.

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,43 @@
+import { StefanLog } from '../../../models';
+
+describe('Login Test', () => {
```
You don't need `* Test` suffix here, since we're in the test file (`*.spec.ts`) :)

You can only add a more descriptive description. Ex. **Login Page** or something.

As a reference, you can use: https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress

sjova commented on cypress/integration/dib-playground/stefan.nikolic_tests/login_test.spec.ts:
```diff
@@ -0,0 +1,43 @@
+    cy.visit('/');
+  });
+
+  it('Empty Login!', () => {
```
each `it` should better describe the test.

Examples:

`should ... something`
`display something`
`navigates to ...`
`submits ...`
`shows something`
`creates something`

As a reference, you can use: https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress

sjova commented on cypress/models/index.ts:
```diff
@@ -1 +1,3 @@
 export * from './user';
+export * from './login(Stefan)';
```
Please never use brackets in the file name. Please stick with the dash case.

sjova commented on cypress/fixtures/stefan_nikolic_data.json:
```diff
@@ -0,0 +1,11 @@
+{
```
Instead `stefan_nikolic_data.json ` please use descriptive file name. And use dash case.

sjova reviewed (commented):
The latest changes look good. I added few more minor improvements (these will produce long-term better code).

You have two errors in your environment. This stuff should be double-checked and fixed before you push PR changes.

Also, you didn't pull the latest changes from the `main` branch.

And remains use better selectors, as we discussed.

---

As a reference you can use:
https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress
https://github.com/cypress-io/cypress-example-kitchensink/tree/master/cypress
https://github.com/cypress-io/cypress-example-recipes

stefannikolic commented:
> The latest changes look good. I added few more minor improvements (these will produce long-term better code).
> 
> You have two errors in your environment. This stuff should be double-checked and fixed before you push PR changes.
> 
> Also, you didn't pull the latest changes from the `main` branch.
> 
> And remains use better selectors, as we discussed.
> 
> As a reference you can use:
> https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress
> https://github.com/cypress-io/cypress-example-kitchensink/tree/master/cypress
> https://github.com/cypress-io/cypress-example-recipes

I will update selectors and fix all that you suggested
